### PR TITLE
fix(assert): module is incompatible with jest@27

### PR DESCRIPTION
**Pull request by cmdcarini**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/assert/package.json [hunk 1]
```diff
@@ -53,7 +53,7 @@
   "peerDependencies": {
     "@aws-cdk/core": "0.0.0",
     "constructs": "^3.3.69",
-    "jest": "^26.6.3"
+    "jest": "^26.6.3 || ^27.0.6"
```
Comment (nija-at) [inline packages/@aws-cdk/assert/package.json:56]: Should probably just be -

```suggestion
    "jest": ">=26.6.3"
```

**Review by nija-at — CHANGES_REQUESTED** [review]

File: packages/@aws-cdk/assert/package.json (hunk 1 above)
Comment (cmdcarini) [reply packages/@aws-cdk/assert/package.json]: Good point, updated

File: packages/@aws-cdk/assert/package.json [hunk 2]
```diff
@@ -53,7 +53,7 @@
   "peerDependencies": {
     "@aws-cdk/core": "0.0.0",
     "constructs": "^3.3.69",
-    "jest": "^26.6.3"
+    "jest": ">=26.6.3"
```
Comment (nija-at) [inline packages/@aws-cdk/assert/package.json:56]: After discussing with @RomainMuller, I take my previous comment back. The correct value here should be

```suggestion
    "jest": ">=26.6.3 <28.0.0"
```

**Review by nija-at — CHANGES_REQUESTED** [review]

File: packages/@aws-cdk/assert/package.json (hunk 2 above)
Comment (cmdcarini) [reply packages/@aws-cdk/assert/package.json]: Applied

**Review by nija-at — DISMISSED** [review]
Thanks for fixing this up.

**Review by nija-at — APPROVED** [review]

**Pull request merged**